nym-network-monitor/src/accounting.rs

```diff
@@ -384,17 +424,73 @@ async fn submit_gateway_stats_to_db(client: Arc<Client>) -> anyhow::Result<()> {
     Ok(())
 }
 
-pub async fn submit_metrics_to_db(client: Arc<Client>) -> anyhow::Result<()> {
+async fn db_connection(database_url: Option<&String>) -> Result<Option<(Client, JoinHandle<()>)>> {
+    if let Some(database_url) = database_url {
+        let (client, connection) = tokio_postgres::connect(database_url, NoTls).await?;
+
+        let handle = tokio::spawn(async move {
+            if let Err(e) = connection.await {
+                error!("Postgres connection error: {}", e);
+            }
+        });
+
+        Ok(Some((client, handle)))
+    } else {
+        Ok(None)
+    }
+}
+pub async fn submit_metrics_to_db(database_url: Option<&String>) -> anyhow::Result<()> {
+    if let Some((client, handle)) = db_connection(database_url).await? {
+        let client = Arc::new(client);
+        let client2 = Arc::clone(&client);
+        let client3 = Arc::clone(&client);
+        submit_node_stats_to_db(client).await?;
+        submit_gateway_stats_to_db(client2).await?;
+        submit_accounting_routes_to_db(client3).await?;
+        handle.abort();
+    }
+    Ok(())
+}
+
+async fn submit_accounting_routes_to_db(client: Arc<Client>) -> anyhow::Result<()> {
     let client = Arc::clone(&client);
-    let client2 = Arc::clone(&client);
-    submit_node_stats_to_db(client).await?;
-    submit_gateway_stats_to_db(client2).await?;
+    let network_account = NetworkAccount::finalize()?;
+    let accounting_routes = network_account.accounting_routes;
+
+    let sink = client
+        .copy_in("COPY routes (layer1, layer2, layer3, gw, success) FROM STDIN BINARY")
+        .await?;
+
+    let writer = BinaryCopyInWriter::new(
+        sink,
+        &[Type::INT4, Type::INT4, Type::INT4, Type::INT4, Type::BOOL],
+    );
+    pin_mut!(writer);
+
+    for route in accounting_routes {
+        writer
+            .as_mut()
+            .write(&[
+                &(route.mix_nodes.0 as i32),
+                &(route.mix_nodes.1 as i32),
+                &(route.mix_nodes.2 as i32),
+                &(route.gateway_node as i32),
+                &route.success,
+            ])
+            .await?;
+    }
+
+    writer.finish().await?;
+
     Ok(())
 }
 
-pub async fn submit_metrics(client: Option<Arc<Client>>) -> anyhow::Result<()> {
-    if let Some(client) = client {
-        submit_metrics_to_db(client).await?;
+pub async fn submit_metrics(database_url: Option<&String>) -> anyhow::Result<()> {
+    match submit_metrics_to_db(database_url).await {
```

Comment:
nit: (because i reckon clippy will complain about it), you could have instead used the `if Err(err) = submit_metrics_to_db(...)` pattern